pravega-cli/src/test/java/io/pravega/tools/pravegacli/SecureControllerCommandsTest.java

```diff
@@ -0,0 +1,170 @@
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class SecureControllerCommandsTest {
+    // Security related flags and instantiate local pravega server.
+    private static final boolean restEnabled = true;
+    private static final boolean authEnabled = true;
+    private static final boolean tlsEnabled = true;
+    private static final Integer restServerPort = 9091;
+    private static LocalPravegaEmulator localPravega;
+    private static final AtomicReference<AdminCommandState> STATE = new AtomicReference<>();
+
+    @Before
+    public void setUp() throws Exception {
+        // Create the secure pravega server to test commands against.
+        LocalPravegaEmulator.LocalPravegaEmulatorBuilder emulatorBuilder = LocalPravegaEmulator.builder()
+                .controllerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .segmentStorePort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .zkPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .restServerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .enableRestServer(restEnabled)
+                .enableAuth(authEnabled)
+                .enableTls(tlsEnabled)
+                .restServerPort(restServerPort);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile("src/test/resources/passwd")
+                    .userName(SecurityConfigDefaults.AUTH_ADMIN_USERNAME)
+                    .passwd(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        }
+        if (tlsEnabled) {
+            emulatorBuilder.certFile("src/test/resources/server-cert.crt")
+                    .keyFile("src/test/resources/server-key.key")
+                    .jksKeyFile("src/test/resources/server.keystore.jks")
+                    .jksTrustFile("src/test/resources/client.truststore.jks")
+                    .keyPasswordFile("src/test/resources/server.keystore.jks.passwd");
+        }
+
+        localPravega = emulatorBuilder.build();
+
+        // Set the CLI properties.
+        STATE.set(new AdminCommandState());
+        Properties pravegaProperties = new Properties();
+        pravegaProperties.setProperty("cli.controllerRestUri", "localhost:" + restServerPort.toString());
+        pravegaProperties.setProperty("pravegaservice.zkURL", localPravega.getInProcPravegaCluster().getZkUrl());
+        pravegaProperties.setProperty("pravegaservice.containerCount", "4");
+        pravegaProperties.setProperty("cli.authEnabled", "true");
```

Comment:
These config names would need to change to meet the new config naming guidelines as mentioned earlier.

Reply:
These changes will be made in a new PR as mentioned by @RaulGracia